aspnet/migration/http-modules.rst

```diff
@@ -1,8 +1,584 @@
+		which is a series events fired by ASP.NET
+		- 
+		`BeginRequest <https://msdn.microsoft.com/en-us/library/system.web.httpapplication.beginrequest(v=vs.100).aspx>`_,
+		`AuthenticateRequest <https://msdn.microsoft.com/en-us/library/system.web.httpapplication.authenticaterequest(v=vs.100).aspx>`_, etc.
+		Each module can create a handler for one or more events;
+
+	2. When handling the same event, the order in which they are configured in the web.config file.
+
+In addition to modules, you can add handlers for the life cycle events to your *Global.asax.cs* file. These handlers run after the handlers in the configured modules.
+
+From handlers and modules to middleware
+---------------------------------------
+ASP.NET 5 middleware is simpler than ASP.NET 4's modules and handlers:
+    * Modules, handlers, Global.asax.cs, web.config (except for IIS configuration) and the application life cycle are gone;
+    * The roles of both modules and handlers have been taken over by middleware;
+    * Middleware is configured using code instead of in the web.config file;
+    * `Pipeline branching <../fundamentals/middleware.html#run-map-and-use>`_ lets you send requests to specific middleware, based on not only the URL but also on request headers, etc.
+
+Middleware is very similar to modules:
+	* Invoked in principle for every request;
+	* Able to stop further processing of a request;
+	* Able to add to the HTTP response, or create their own.
+
+Middleware and modules are processed in a different order:
+	* While the request is going up the pipeline, middleware is processed in the order in which it is inserted into the request pipeline, similar to modules;
+	* But while the response is going back to the browser, middleware is processed in the reverse order.
+
+.. image:: http-modules/_static/middleware.png
+
+Migrating module code to middleware
+------------------------------------
+In your ASP.NET 4 application, your module will look similar to this:
+
+.. literalinclude:: http-modules/sample/Asp.Net4/Asp.Net4/Modules/MyModule.cs
+	:language: c#
+	:linenos:
+	:emphasize-lines: 6, 8, 24, 31
+
+As shown in the :ref:`introduction to middleware <fundamentals-middleware>`,
+an ASP.NET 5 middleware is simply a class that exposes an ``Invoke`` method taking an ``HttpContext`` and returning a ``Task``. Your new middleware will look like this:
+
+.. _http-modules-usemiddleware:
+
+.. literalinclude:: http-modules/sample/Asp.Net5/src/Asp.Net5/Middleware/MyMiddleware.cs
+	:language: c#
+	:linenos:
+	:emphasize-lines: 9, 13, 20, 24, 28, 30, 32
+
+The middleware template used here is the same as that shown in the section on 
+`writing middleware <../fundamentals/middleware.html#writing-middleware>`_.
+
+Towards the bottom is a helper class to make it easier to configure your middleware in your ``Startup`` class. 
+This calls the ``UseMiddleware`` method, which instantiates your middleware class, dependency injecting all objects required by 
```

Comment:
> This calls the ``UseMiddleware`` method, which instantiates your middleware class, dependency injecting all objects 

The \`\`UseMyMiddleware\`\` method adds your middleware class to the request pipeline. Services required by the middleware get injected in the middleware's constructor.